### DISCUSSION TITLE
phase required error

### DISCUSSION THREAD

**Discussion by joelyss** [QSMxT/QSMxT#32] [Q&A] [answered]
hi,

firstly - many thanks for all the hard work that has gone into neurodesktop and QSMxT - it seems like a really great resource. i followed your ISMRM talks regarding NeuroDesktop and QSM and found them extremely helpful and interesting.

i am quite new to NeuroDesktop and to QSMxT software - i have previously run software that i have d/l on my windows UBUNTU shell or on MacOS - so apologies my error is probably from a basic misunderstanding.

i have installed docker and neurodesktop no problem. moved my dicom data to the storage folder which is 13 dicom folders: 1 T1, 6 echoes of mag and phase dicoms each.

i followed the tutorial and have been trying to apply the same to my data, however i think there might be an issue during the bids conversion as once i try to run 'run_2_qsm' i get the following error:
tgv_qsm: error: argument -p/--phase required

guidance appreciated - or even just pointing me towards some useful resources as my searching has not been very fruitful
many thanks,
joely

**Comment by stebo85** [comment]
Dear @joelyss 

my first guess is that the automatic dicom to bids conversion doesn't detect your GRE data as QSM and that's one reason why you see the error.

There are a few ways how we can fix this, but it would be important to know which version of QSMxT you are currently running?

Can you also list the content of the 01_bids/ directory?

If you like, you can also send an example of your dicom data (s.bollmann@uq.edu.au) and we can have a look why it's not being recognised. Alternatively we can have a quick zoom call to figure out what's happening on your side.

Cheers
Steffen

**Comment by joelyss** [comment]
Hi Steffen,

Thank you for your response

I am running QSMxT 1.1.6 via NeuroDesktop. Inside the bids folder is the following:

code > bidscoin > bidscoiner.errors, bidscoinder.log, bidsmap.yaml, bidsmapper.errors, bidsmapper.log
sub > ses-1 > sub_ses-1_scans.tsv
sub > ses-1 > anat > sub_ses-1_acq-T1w_run-1_magnitude.json
sub > ses-1 > anat > sub_ses-1_acq-T1w_run-1_magnitude.nii.gz
dataset_description.json
participants.json
participants.tsv
README 

So I have two options of the data format. The first - for me is the simpliest - which is the anonymised data from the Siemens scanner sorted by series description into dicom folders (.IMA). This has all echoes in a single folder. However, when I tried to run QSMxT on that data it did not recognise the data at all, and I couldn't get past the first step of processing. 

So then I instead took the data from our PACS/RIS system and anonymised. This has all the echoes in separate folders (without an extension, so I added .IMA). This appeared to work better but it still has the issue described above.

A zoom would be great if you have availability

Many thanks,
Joely

**Reply by stebo85** [reply]
Dear @joelyss,

Yes, it really doesn't recognize your GRE files. We made a few improvements in the last versions to the dicom sorting. Can you try the current version 1.1.9? It is included in the current neurodesk version https://neurodesk.github.io/

Cheers
Steffen

**Reply by joelyss** [reply]
I have reinstalled neurodesk and still it is version 1.1.6 f QSMxT

**Reply by stebo85** [reply]
which version of neurodesk did you install? neurodesktop 20220128 is the current version and it contains qsmxt 1.1.9 

Send me quick email with your availability if you like a quick zoom call to get you started: s.bollmann@uq.edu.au

**Reply by joelyss** [reply]
Great thank you, I will email you.

I am following the link: https://neurodesk.github.io/
Then selecting Windows, so I am taking the version from this link:
https://github.com/NeuroDesk/neurodesktop/raw/main/Windows_run_Neurodesk/NeuroDesktop.exe

**Reply by stebo85** [reply]
AHH, that's a bug in the exe file and it wasn't updated to the latest version (@thomshaw92 - are you on this?). 

Can you use the normal start-up command for Windows until the bug is fixed?

**Reply by stebo85** [reply]
Dear @joelyss ,

Thanks to your bug report, the neuroDestkop.exe is now updated and downloads the latest neurodestop version with QSMxT 1.1.9 included. Let us know how this works for you :)

Cheers
Steffen

**Reply by joelyss** [reply]
Apologies for my delayed response - I was OOO yesterday

This is great thank you - I will update neurodesktop and retry with my data today and report back

**Reply by joelyss** [reply]
Hi Steffen,

So I have successfully updated to NeuroDesktop with QSMxT v 1.1.9 - many thanks!

However, I continue to get errors. The error now is that it does not recognised the dicoms.

I run the first command e.g.
python3 /opt/QSMxT/run_0_dicomSort.py Desktop/neurodesktop-storage/folder1 00_dicom

And I get the following error:
reading file list...
0 dicom files found.
sorting dicoms in 00_dicom...
done sorting dicoms.

Checking the 00_dicoms folder created it is empty. I have tried this on .IMA and .dcm data, and on both the Siemens anonymised data and the PACS data.

Many thanks,
Joely

**Reply by stebo85** [reply]
Dear @joelyss,

can you send me an example of the data (s.bollmann@uq.edu.au) so I can check why it's not being recognised? 

Cheers
Steffen

**Comment by stebo85** [comment, marked as answer]
Problems identified and solved:
1) SeriesDescription does not exist anymore in newer dicoms -> replaced with ProtocolName because it exists in older files and new files.
2) bugfix now allows dots in protocol names (e.g. mp2rage_0.75_mm)
